### ISSUE TITLE
Fix and re-run irrigation extractor

### ISSUE THREAD

**Issue by craig-willis** [terraref/computing-pipeline#400]
Opted to create a new ticket for this since it's really separate from  https://github.com/terraref/reference-data/issues/196.

Fix and re-run the irrigation extractor:

Completion criteria:
* [x] Fix problem (what's coming out of pipe), implement change, 
* [x] review/merge PR
* [x] ~~Get irrigation polygons and figure out what to do with it (i.e., BETY treatment).  Need to have documentation addressing what to do with liters/second values. Need to account for which polygons are associated with which ranges of time.~~
* [x] Purge geostreams of irrigation data
* [x] Build docker container
* [ ] deploy, re-run 
* [x] Review outputs

**Comment by dlebauer** [comment]
Renamed irrigation_flux (kg/m2/s) to irrigation_transport (kg/s) following cf vocabulary 

The first step: fix problem should be resolved by two PRs:
* https://github.com/terraref/extractors-environmental/pull/25
* https://github.com/terraref/terrautils/pull/42

**Comment by dlebauer** [comment]
Draft queries to define irrigation zones are here: https://gist.github.com/dlebauer/28d9b22b7c4c88c35f96ae167aa45e71 waiting on replies to https://github.com/terraref/reference-data/issues/196#issuecomment-360920825

**Comment by max-zilla** [comment]
Merged terrautils PR.

**Comment by max-zilla** [comment]
@dlebauer should update pull request based on Wei's suggestions in extractors-environmental.

**Comment by max-zilla** [comment]
@dlebauer I was attempting to finalize this PR. however, the cfunits python library has a dependency on the C library UDUNITS:

I add this to the Dockerfile:
```
RUN     cd /usr/local/src && \
        wget ftp://ftp.unidata.ucar.edu/pub/udunits/udunits-2.2.24.tar.gz && \
        gunzip -c udunits-2.2.24.tar.gz   | pax -r && \
        cd udunits-2.2.24 && ./configure && make install && make clean
```

With that + pip install cfunits a bug is encountered when trying to import cfunits:
```
File "/usr/local/lib/python2.7/dist-packages/cfunits/units.py", line 216, in <module>
    assert(0 == _ut_unmap_symbol_to_unit(_ut_system, _c_char_p('Sv'), _UT_ASCII))
```

I spent a few minutes trying to understand why this was failing. 
https://bitbucket.org/cfpython/cfunits-python/src/ac37a81ede9b54ff8f4256909d03c1300056ea7f/cfunits/units.py?at=master&fileviewer=file-view-default

Here is the comment in this portion of the cfunits script:
```
# Change Sv mapping. Both sievert and sverdrup are just aliases,
# so no unit to symbol mapping needs to be changed.
# We don't need to remove rem, since it was constructed with
# the correct sievert mapping in place; because that mapping
# was only an alias, the unit now doesn't depend on the mapping
# persisting.
assert(0 == _ut_unmap_symbol_to_unit(_ut_system, _c_char_p('Sv'), _UT_ASCII))
assert(0 == _ut_map_symbol_to_unit(_c_char_p('Sv'), _UT_ASCII,
                                   _ut_get_unit_by_name(_ut_system, _c_char_p('sverdrup'))))
```

...and a bit more definition of these parameters:
```
assert(0 == _ut_unmap_symbol_to_unit(_ut_system, _c_char_p('Sv'), _UT_ASCII))

_ut_unmap_symbol_to_unit = _udunits.ut_unmap_symbol_to_unit
_ut_unmap_symbol_to_unit.argtypes = (_c_void_p, _c_char_p, _c_int)
_ut_unmap_symbol_to_unit.restype = _c_int
_udunits = ctypes.CDLL('libudunits2.so.0')

_ut_system = _ut_read_xml(None)

_c_char_p      = ctypes.c_char_p
_UT_ASCII      = 0
```

At this point I've spent a fair amount of time trying to get this to work, and during sprint planning Craig suggested bouncing this back to you.
- it seems important to get import cfunits working in python for other purposes as well.
- I'm getting into C library things above that perhaps make more sense to you?

you should be able to build my dockerfile here:
https://github.com/terraref/extractors-environmental/blob/irrigation-comments-cfunits/irrigation_datparser/Dockerfile
```
docker build -f Dockerfile -t irrigation_dockertest .
docker run irrigation_dockertest
```
...and encounter the bug above. Could you take a few minutes to see if you can update the Dockerfile installation process so this cfunits import doesn't fail in python? Or if you don't wish to take the time, we discuss how to revert away from cfunits to something else.

**Comment by max-zilla** [comment]
consider if there is another library besides cfunits if we can't get cfunits piece properly.

**Comment by max-zilla** [comment]
try ```apt-get install udunits-bin libudunits2-dev```

**Comment by max-zilla** [comment]
The new apt-get install worked!

```
>>> from cfunits import Units
>>> def gallon2liter(value):
     if value:
         return Units.conform(int(value), 
                              Units('gallon / day'),
                              Units('liter / second'), inplace=True)
     return 0.0
 
>>> gallon2liter(20)
0.0008762527777777775
```

20 gallons/day * ~3.78541 liters/gallon = 75.7082 liters/day
75.7082 liters/day / 86400 seconds/day =  ~0.000876252314815 liters/second

**Comment by max-zilla** [comment]
Purging datapoints: on terra-postgres, log into geostream db.
```delete from datapoints where stream_id = 3212;```

To submit irrigation collection to the extractor:
```
from pyclowder.files import submit_extractions_by_collection
from pyclowder.connectors import Connector
c = Connector({})
submit_extractions_by_collection(c, "http://terraref.ncsa.illinois.edu/", "SECRET_KEY", "58ed660b4f0c430e2ea7812a", "terra.environmental.irrigation_datparser")
```

**Comment by max-zilla** [comment]
https://terraref.ncsa.illinois.edu/clowder/api/geostreams/datapoints?stream_id=3212

**Comment by max-zilla** [comment]
@dlebauer please review the new datapoints and we can close.

**Comment by max-zilla** [comment]
@dlebauer

**Comment by max-zilla** [comment]
i will send @dlebauer link to verify these data.

**Comment by dlebauer** [comment]
it is ready to re-run on all data
